src/Entity/User.php

```diff
@@ -15,47 +15,56 @@ class User implements UserInterface
     private $id;
 
     /**
-     * @var string 
+     * @var string
      */
     private $email;
-    
+
     /**
-     * @var string 
+     * @var string
      */
     private $username;
-    
+
     /**
-     * @var string 
-     */    
+     * @var string
+     */
     private $password;
 
     /**
-     * @var UserCourse 
-     */        
+     * @var array
+     */
+    private $roles;
+
+    /**
+     * @var UserCourse[]
+     */
     private $viewedCourses;
-    
+
     /**
      * @param string $email
      * @param string $username
      * @param string $password
+     * @parma array $roles
      */
-    private function __construct(string $email, string $username, string $password)
+    public function __construct(string $email, string $username, string $password, array $roles)
     {
         $this->email = $email;
         $this->username = $username;
         $this->password = $password;
+        $this->roles = $roles;
+        $this->viewedCourses = new ArrayCollection();
     }
-    
+
     /**
      * @param string $email
      * @param string $username
      * @param string $password
-     */    
-    public static function register(string $email, string $username, string $password)
+     * @parma array $roles
+     */
+    public static function register(string $email, string $username, string $password, array $roles)
```

Comment:
Maybe this `register` name is irrelevant as no DB write is done here. Maybe you could rename this fn `create` or `build`, as it is more in accordance with what the function does : building a new `User` object ;)